nwbinspector/checks/tables.py

```diff
@@ -74,6 +75,39 @@ def check_time_intervals_stop_after_start(time_intervals: TimeIntervals, nelems:
         )
 
 
+@register_check(importance=Importance.BEST_PRACTICE_SUGGESTION, neurodata_type=DynamicTable)
+def check_column_binary_capability(table: DynamicTable, nelems: int = 200):
+    """
+    Check each column of a table to see if the data could be set as a boolean dtype.
+
+    Parameters
+    ----------
+    time_intervals: TimeIntervals
+    nelems: int
+        Only check the first {nelems} elements. This is useful in case there columns are
+        very long so you don't need to load the entire array into memory. Use None to
+        load the entire arrays.
+    """
+    for column in table.columns:
+        if hasattr(column, "data"):
+            column_data = column.data if hasattr(column.data, "dtype") else np.array(column.data)
+            unique_values = np.unique(column_data[:nelems])
+            column_dtype = column_data.dtype
+            saved_bytes = (column_dtype.itemsize - 1) * column_data.size
+            if (
+                column_data.dtype != np.dtype("object")  # jagged shaped lists
+                and len(column_data.shape) == 1
```

Comment:
Why does it matter if the data is a 1D vector? This test seems applicable to columns of any shape.

Reply:
The question is, should it be the responsibility of this check to examine the set of values from all nested array contents?

I was operating under the assumption that it would only operate on simple 1D vectors.

Reply:
> The question is, should it be the responsibility of this check to examine the set of values from all nested array contents?

what do you mean?

Reply:
Right now it only inspects the contents of columns that contain a single value, such as https://github.com/NeurodataWithoutBorders/nwbinspector/pull/72/files#diff-c5a67d4b9c1f4fdf1c7808b5f06bdea82886bd08a3700d3073a79008253b86b2R164-R182

It then bypasses the more complicated settings where the value at each row is itself a vector: https://github.com/NeurodataWithoutBorders/nwbinspector/pull/72/files#diff-c5a67d4b9c1f4fdf1c7808b5f06bdea82886bd08a3700d3073a79008253b86b2R132-R153

Reply:
I think it would be better to check the target data values in the case of a VectorIndex like I am suggesting above